code/game/objects/items/weapons/dnascrambler.dm

```diff
@@ -4,44 +4,37 @@
 	icon = 'icons/obj/syringe.dmi'
 	item_state = "syringe_0"
 	icon_state = "b10"
-	var/used = null
 
-	update_icon()
-		if(used)
-			icon_state = "b0"
+/obj/item/weapon/dnascrambler/attack(mob/M as mob, mob/user as mob)
+	if(!M || !user)
+		return
+
+	if(!ishuman(M) || !ishuman(user))
+		return
+
+	if(M == user)
+		user.visible_message("<span class='danger'>[user.name] injects \himself with [src]!</span>")
+		src.injected(user,user)
+	else
+		user.visible_message("<span class='danger'>[user.name] is trying to inject [M.name] with [src]!</span>")
+		if (do_mob(user,M,30))
+			user.visible_message("<span class='danger'>[user.name] injects [M.name] with [src].</span>")
+			src.injected(M, user)
 		else
-			icon_state = "b10"
+			to_chat(user, "<span class='warning'>You failed to inject [M.name].</span>")
 
-	attack(mob/M as mob, mob/user as mob)
-		if(!M || !user)
-			return
+/obj/item/weapon/dnascrambler/proc/injected(var/mob/living/carbon/target, var/mob/living/carbon/user)
+	target.generate_name()
+	target.real_name = target.name
 
-		if(!ishuman(M) || !ishuman(user))
-			return
+	scramble(1, target, 100)
 
-		if(src.used)
-			return
+	log_attack("[key_name(user)] injected [key_name(target)] with the [name]")
+	log_game("[key_name_admin(user)] injected [key_name_admin(target)] with the [name]")
 
-		if(M == user)
-			user.visible_message("<span class='danger'>[user.name] injects \himself with [src]!</span>")
-			src.injected(user,user)
-		else
-			user.visible_message("<span class='danger'>[user.name] is trying to inject [M.name] with [src]!</span>")
-			if (do_mob(user,M,30))
-				user.visible_message("<span class='danger'>[user.name] injects [M.name] with [src].</span>")
-				src.injected(M, user)
-			else
-				to_chat(user, "<span class='warning'>You failed to inject [M.name].</span>")
-
-	proc/injected(var/mob/living/carbon/target, var/mob/living/carbon/user)
-		target.generate_name()
-		target.real_name = target.name
-
-		scramble(1, target, 100)
-
-		log_attack("[key_name(user)] injected [key_name(target)] with the [name]")
-		log_game("[key_name_admin(user)] injected [key_name_admin(target)] with the [name]")
+	//We don't want to leave an obvious "used dna scrambler" behind, let's just create a harmless syringe.
+	var/obj/item/weapon/reagent_containers/syringe/spent = new(get_turf(user))
```

Comment:
Are spent syringes the "broken" ones? Doesn't seem like this is any less obvious considering it spawns on the floor instead of in the hand of the user.